Readme.md

```diff
@@ -1,32 +1,129 @@
 
-You can create your own grimore and be part of the DataWeave Wizards Secret Club.
+You can create your own grimoire and be part of the DataWeave Wizards Secret Club.
 
-## How to create a SPELL
+Let's start defining some concepts:
 
-- wizard-name = Github User Name
+## Wizard
+A `wizard` is just a name that we trust and we want to have all their `spells` available.
 
-All you need is 
- - A repo `*<wizard-name>*-data-weave-grimoire`
- - A folder that is going to be your spell name i.e HelloWorld/
- - Then a file Main.dwl that needs to be under <spell_name>/src/Main.dwl in our case HelloWorld/src/Main.dwl
- - And then write your script inside Main.dwl for example:
+A `wizard` has a `grimoire`, a GitHub repository that is inferred by `https://github.com/${wizard_name}/${wizard_name}-data-weave-grimoire`, that includes all the `spells` from this `wizard`.
+
+## Grimoire
+
+A `grimoire` is a collection of `spells` from a given `wizard`. 
+
+## Spells
+
+A `spell` is a very simple way to distribute and share a DataWeave transformation.
+
+A `spell` is just executables scripts that can be called from the command-line using the `spell` name.
+
+## DataWeave Grimoire
+
+DataWeave Grimoire is the default grimoire provided by DataWeave CLI.
+
+This grimoire contains the following spells:
+
+| Spell        | Description                                                        |
+|--------------|--------------------------------------------------------------------|
+| HelloWizard  | Just a simple example that shows how to create a very basic spell. |
+| CountWizards | Spell to find other DataWeave wizards.                             |
+| Playground   | This spell will launch the playground UI.                          |
+
+## How to Add a Wizard
+
+In order to add a new `wizard` use the DataWeave CLI
+
+```bash
+dw --add-wizard <wizard_name>
 ```
-  %dw 2.0
-  ---
-  "Hi Wise Wizard" 
+The wizard grimoire is going to be cloned at `{user.home}/.dw/grimoires`
+
+:warning: **Do NOT add a wizard you don't trust**
+
+## Listing All the Available Spells
+
+Using the `--list-spells` it will show all the available spells for each wizard with the documentation of each spell.
+
+```bash
+dw --list-spells
 ```
 
+## How to Run a Spell
+
+There are several ways to run a spell using the DataWeave CLI.
+
+:warning: **Do NOT execute spells you don't trust**
 
-Then you go to your command line and type `dw --spell <wizard-name>/HelloWorld`
+### Running a Spell From a Wizard
 
-### WARNING: Do NOT execute spells you don't trust
+```bash
+dw --spell <wizard_name>/<spell_name>
+```
 
-## How to test Spells locally before pushing to a repo
+### Running a Spell From the DataWeave Grimoire
 
-Just run your dwl script with:
+```bash
+dw --spell <spell_name>
 ```
-dw -f <path_to_main_script>
+
+### Running a Spell From a Grimoire a Local Folder
```

Comment:
```suggestion
### Running a Spell From a Local Grimoire
```

Reply:
I'm not sure if this was the meaning. Does it mean a grimoire you have somewhere besides `.dw/grimoires`?

Reply:
This command runs the spell based on your current path. I added a few more references.